src/main/java/duke/manager/TaskManager.java

```diff
@@ -0,0 +1,197 @@
+package duke.manager;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+
+public class TaskManager {
+
+    /**
+     * Prints a line on the console
+     */
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Determines the task type,
+     * assuming it is the first
+     * word in the message and
+     * returns it
+     *
+     * @param message The input by the user.
+     * @return The first word of the string.
+     */
+    private static String taskType(String message) {
+        String[] type = message.split(" ");
+        return type[0];
+    }
+
+
+    /**
+     * The function adds the task input by the user
+     *
+     * @param tasks   the array of tasks
+     * @param message the input string containing
+     *                the task description
+     */
+    private static void addTask(Task[] tasks, String message) {
+        switch (taskType(message)) {
+        case "todo":
+            addTodo(tasks, message);
+            break;
+        case "deadline":
+            addDeadline(tasks, message);
+            break;
+        case "event":
+            addEvent(tasks, message);
+            break;
+        default:
+            printLine();
+            System.out.println("I really cannot understand what you wrote");
+            printLine();
+            // Fallthrough
+        }
+        if (Task.COUNT > 0) {
+            printAddedTask(tasks[Task.COUNT - 1]);
+        }
+    }
+
+    /**
+     * The function prints the recently added task
+     */
+    private static void printAddedTask(Task task) {
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task.getDescription());
+        System.out.println("Now you have " + Task.COUNT + " tasks in the list.");
+        printLine();
+    }
+
+    /**
+     * The function adds the todo task input by the user
+     *
+     * @param tasks   the array of tasks
+     * @param message the input string containing
+     *                the task description
+     */
+    private static void addEvent(Task[] tasks, String message) {
+        try {
+            message = message.substring(6);
+            String[] eventData = message.split("/", 2);
+            tasks[Task.COUNT] = new Event(eventData[0], eventData[1].substring(3));
+            Task.COUNT++;
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("YOU IDIOT !!??!! The description of an event cannot be empty.");
```

Comment:
I thinking the StringIndexOutOfBoundsExceptions caught in this case might be caused by problems other than the "description is empty" problem. Also, the message showed to the user might contain language not so appropriate. Similar problems can also be found in later exception handling.